pkg/core/aliaslist.go

```diff
@@ -0,0 +1,133 @@
+package core
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	"github.com/zinclabs/zinc/pkg/metadata"
+	"github.com/zinclabs/zinc/pkg/zutils"
+)
+
+var ZINC_INDEX_ALIAS_LIST AliasList
+
+type AliasList struct {
+	lock    sync.RWMutex
+	Aliases map[string][]string
+}
+
+func NewAliasList() *AliasList {
+	return &AliasList{Aliases: map[string][]string{}}
+}
+
+func (al *AliasList) AddIndexesToAlias(alias string, indexes []string) error {
+	al.lock.Lock()
+	al.Aliases[alias] = append(al.Aliases[alias], indexes...)
+
+	err := metadata.Alias.Set(al.Aliases)
+	if err != nil {
+		log.Err(err).Msg("failed to save alias in metadata after add operation")
+		al.lock.Unlock()
+		return err
+	}
+
+	al.lock.Unlock()
+	return nil
+}
+
+func (al *AliasList) RemoveIndexesFromAlias(alias string, removeIndexes []string) error {
+	al.lock.Lock()
+
+	indexes, ok := al.Aliases[alias]
+	if !ok {
+		al.lock.Unlock()
+		return nil
+	}
+
+outer:
```

Comment:
And break `tag` is a magic method, can we use other way replace it?

Reply:
Not sure, do you have an idea?

Reply:
I have an idea, no need copy data every time and delete the tag, we can do it like this:

```
removeIndexesMap:=make(map[string]bool)
for _,index:=range removeIndexes {
    removeIndexesMap[index] = true
}
lastIndex:=len(indexes);
for i:=0; i < lastIndex; i++ {
    if _,ok:= removeIndexesMap[lindexes[i]]; ok {
             indexes[lastIndex], indexes[i] = indexes[i], indexes[lastIndex]
             i--
             lastIndex--
    }
}

indexes = indexes[:lastIndex]
```